### PR TITLE
[POC] implement test_arithmetic.py

### PR DESCRIPTION
**Pull request by jbrockmendel**
There are a ton of scattered arithmetic tests for Index/Series/DataFrame that _should_ be testing the same things, but in fact are haphazard.  Fixing this given the current structure would entail an enormous about of code duplication.

This PR if a proof of concept for gathering all those tests in one test_arithmetic.py file, parametrizing them, and ensuring that the relevant behavior is identical across arraylike classes.

As Datetime/Timedelta/Period EA come online, the case for de-duplication will be even stronger.

In this form it is really easy to track (via xfails) what behavior needs fixing.

### CODE REVIEW

**Comment by gfyoung** [comment]
@jbrockmendel : So even though this is a POC, you still want to merge this, right?

**Comment by jbrockmendel** [comment]
If there is consensus about this goal, yes.

**Comment by jbrockmendel** [comment]
@jreback related to discussion about test refactoring.

**Comment by jorisvandenbossche** [comment]
The idea would be to add the arrays as when they exist?

general idea looks good to me

File: pandas/tests/test_arithmetic.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import pytest
+import numpy as np
+
+import pandas as pd
+import pandas.util.testing as tm
+
+from pandas import Timedelta
+
+
+def assert_equal(left, right, **kwargs):
```
Comment (jreback) [inline pandas/tests/test_arithmetic.py]: move to utils.testing

File: pandas/tests/test_arithmetic.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        tm.assert_index_equal(left, right, **kwargs)
+    elif isinstance(left, pd.Series):
+        tm.assert_series_equal(left, right, **kwargs)
+    elif isinstance(left, pd.DataFrame):
+        tm.assert_frame_equal(left, right, **kwargs)
+    else:
+        raise NotImplementedError(type(left))
+
+
+def box_expected(expected, box_cls):
```
Comment (jreback) [inline pandas/tests/test_arithmetic.py]: maybe put in conftest.py, these are test helper functions that we don't necessarily want to publicly expose

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_arithmetic.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/test_arithmetic.py]: alternatively, we should put this into _test_decorators (maybe rename though)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
all for this!

**Pull request merged**